### PR TITLE
refactor: extract invoice expiry calculation into ExpiryUtils

### PR DESCRIPTION
**Pull request by myxmaster**
# Description

Extracts the expiry-seconds calculation logic from `Receive.tsx` and `InvoicesSettings.tsx` into `utils/ExpiryUtils.ts`.

Additionally "CREATE INVOICE" button on Receive screen is now disabled when expiry input is empty (which would pass `NaN` as the invoice expiry and display an error).

This pull request is categorized as a:

- [ ] New feature
- [x] Bug fix
- [x] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance
- [ ] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [x] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] Embedded LND
- [x] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (CLNRest)
- [ ] Nostr Wallet Connect
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the ZEUS [Transfix page](https://app.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

**Comment by myxmaster** [comment]
not sure about geminis suggestions... sure, kind of true, but also a bit unnecessary i guess. your call @kaloudis

**Comment by kaloudis** [comment]
> not sure about geminis suggestions... sure, kind of true, but also a bit unnecessary i guess. your call @kaloudis

I think it really should be the other way around. What the hell do the indexes mean if you're looking at the code at the first time? Can make those enums.

File: utils/ExpiryUtils.test.ts
Comment (kaloudis) [inline utils/ExpiryUtils.test.ts:1]: We're  missing test for edge cases — No tests for empty string, '0', negative values, or non-numeric input to `expirySecondsFromInput`. With BigNumber, passing '' produces 'NaN' which would then be stored as `expirySeconds`. The !expiry guard on the button helps, but the function itself doesn't protect against it.

File: views/Receive.tsx [hunk 1]
```diff
@@ -180,6 +181,30 @@ enum RouteHintMode {
     Custom = 1
 }
 
+enum ExpirationPreset {
```
Comment (kaloudis) [inline views/Receive.tsx]: this could potentially live in `ExpiryUtils`

File: views/Receive.tsx [hunk 2]
```diff
@@ -180,6 +181,30 @@ enum RouteHintMode {
...
+    TenMinutes = 0,
+    OneHour = 1,
+    OneDay = 2,
+    OneWeek = 3
+}
+
+const expirationIndexFromSeconds = (
```
Comment (kaloudis) [inline views/Receive.tsx]: this func could also live in `ExpiryUtils`

File: utils/ExpiryUtils.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const SECONDS_IN_HOUR = SECONDS_IN_MINUTE * 60;
+const SECONDS_IN_DAY = SECONDS_IN_HOUR * 24;
+const SECONDS_IN_WEEK = SECONDS_IN_DAY * 7;
+
+// Converts a user-entered value + time period into expiry seconds.
+// e.g. ('2', 'Hours') → '7200'
+
+export const expirySecondsFromInput = (
+    value: string,
+    timePeriod: string
```
Comment (kaloudis) [inline utils/ExpiryUtils.ts]: do we want to more tightly define the `timePeriod` values here?
